physics/module_SGSCloud_RadPre.F90

```diff
@@ -82,97 +91,207 @@ subroutine sgscloud_radpre_run(    &
-              !Partition the convective clouds into water & ice according to a linear
-              qc(i,k) = qc(i,k)+qci_conv(i,k)*(min(1., max(0., (T3D(i,k)-244.)/25.)))
-              qi(i,k) = qi(i,k)+qci_conv(i,k)*(1. - min(1., max(0., (T3D(i,k)-244.)/25.)))
-
-              Tc = T3D(i,k) - 273.15
-
-              if (nint(slmsk(i)) == 1) then !land
-                if(qc(i,k)>1.E-8)clouds3(i,k)=5.4                !eff radius cloud water (microns)
-                !eff radius cloud ice (microns), from Mishra et al. (2014, JGR Atmos)
-                if(qi(i,k)>1.e-8)clouds5(i,k)=max(173.45 + 2.14*Tc, 20.)
-              else
-                !eff radius cloud water (microns), from Miles et al. 
-                if(qc(i,k)>1.E-8)clouds3(i,k)=9.6
-                !eff radius cloud ice (microns), from Mishra et al. (2014, JGR Atmos, fig 6b)
-                if(qi(i,k)>1.E-8)clouds5(i,k)=max(173.45 + 2.14*Tc, 20.)
+            qc_save(i,k) = qc(i,k)
+            qi_save(i,k) = qi(i,k)
+          end do
+        end do
+
+        if ( do_mynnedmf ) then
+
+          ! add boundary layer clouds - Note: now the temperature-dependent sorting of
+          ! ice and water subgrid-scale clouds is done inside the MYNN-EDMF
+
+          do k = 1, levs
+            do i = 1, im
+
+              !if (imp_physics == imp_physics_gfdl) then
+              !  ! only complement the GFDL cloud fractions
+              !  if (clouds1(i,k) < 0.01 .and. cldfra_bl(i,k) > 0.01) then
+              !    clouds1(i,k) = cldfra_bl(i,k)
+              !  endif
+              !else
+                 clouds1(i,k) = cldfra_bl(i,k)
+              !endif
+
+              !if( qr(i,k) > 1.0e-7 .OR. qs(i,k) > 1.0e-7.or.qci_conv(i,k)>1.0e-7)THEN
+                 !Keep Xu-RandalL clouds fraction - do not overwrite
+              !else
+              !    clouds1(i,k) = cldfra_bl(i,k)
+              !endif
+
+              if (qc(i,k) < 1.e-6 .and. cldfra_bl(i,k)>0.001) then
+                qc(i,k) = qc_bl(i,k)*cldfra_bl(i,k)
+                if (nint(slmsk(i)) == 1) then !land
+                  if(qc(i,k)>1.E-8)clouds3(i,k)=5.4                !eff radius cloud water (microns)
+                else
+                  !eff radius cloud water (microns), from Miles et al.
+                  if(qc(i,k)>1.E-8)clouds3(i,k)=9.6
+                endif
+                !calculate the liquid water path using additional BL clouds
+                clouds2(i,k) = max(0.0, qc(i,k) * gfac * delp(i,k))
               endif
-            endif
+              if (qi(i,k) < 1.e-8 .and. cldfra_bl(i,k)>0.001) then
+                qi(i,k) = qi_bl(i,k)*cldfra_bl(i,k)
+                Tc = T3D(i,k) - 273.15
+                !iwc = qi(i,k)*1.0e6*rho(i,k)
+                if (nint(slmsk(i)) == 1) then !land
+                  !eff radius cloud ice (microns), from Mishra et al. (2014, JGR Atmos, fig 6b)
+                  if(qi(i,k)>1.E-8)clouds5(i,k)=max(173.45 + 2.14*Tc, 20.)
+                else
```

Comment:
Does this "else" apply to everything other than land? Right now, FV3 has "ocean" and "ice", very soon also "lake". If not, let's keep it in mind for future improvements.